acceptance_testing.go

```diff
@@ -1074,27 +1086,65 @@ func (a acceptanceTest) isEqualRecords(is *is.I, want, got []Record) {
 		return
 	}
 
-	// transform slices into maps so we can disregard the order
-	wantMap := make(map[string]Record, len(want))
-	gotMap := make(map[string]Record, len(got))
-	for i := 0; i < len(want); i++ {
-		wantMap[string(want[i].Payload.Bytes())] = want[i]
-		gotMap[string(got[i].Payload.Bytes())] = got[i]
+	a.sortMatchingRecords(want, got)
+
+	if bytes.Equal(want[0].Bytes(), got[0].Payload.After.Bytes()) {
+		// the payload of the actual record matches the whole record in the
+		// expectation, the destination apparently writes records as a whole and
+		// retrieves them as a whole in the payload
+		// this is valid behavior, we need to adjust the expectations
+		for i, wantRec := range want {
+			want[i] = Record{
+				Position:  nil,
+				Operation: OperationSnapshot, // we allow operations Snapshot or Create
+				Metadata:  nil,               // no expectation for metadata
+				Key:       got[i].Key,        // no expectation for key
+				Payload: Change{
+					Before: nil,
+					After:  RawData(wantRec.Bytes()), // the payload should contain the whole expected record
+				},
+			}
+		}
 	}
 
-	is.Equal(len(got), len(gotMap)) // record payloads are not unique
+	for i := range want {
+		a.isEqualRecord(is, want[i], got[i])
+	}
+}
 
-	for key, wantRec := range wantMap {
-		gotRec, ok := gotMap[key]
-		is.True(ok) // expected record not found
+func (a acceptanceTest) sortMatchingRecords(want, got []Record) {
```

Comment:
Nitpick: would be nice to briefly explain what sortMatchingRecords is about, especially since it's "flexible" in sorting.

Reply:
Done!